finality-aleph/src/environment.rs

```diff
@@ -156,56 +156,35 @@ where
     pub(crate) fn new(
         client: Arc<C>,
         select_chain: SC,
+        tx_consensus: mpsc::UnboundedSender<NotificationIn<H>>,
+        rx_consensus: mpsc::UnboundedReceiver<NotificationOut<H>>,
         tx_network: mpsc::UnboundedSender<NetworkCommand<B, H>>,
         rx_network: mpsc::UnboundedReceiver<NetworkEvent<B, H>>,
-        _config: Option<Config>,
-        _authorities: Vec<AuthorityId>,
+        rx_order: tokio::sync::mpsc::UnboundedReceiver<Vec<H>>,
         auth_cryptostore: AuthorityKeystore,
         hashing: impl Fn(&[u8]) -> H + Send + Copy + 'static,
         epoch_id: EpochId,
     ) -> Self {
         Environment {
             client,
             select_chain,
-            tx_consensus: None,
-            rx_consensus: None,
+            tx_consensus,
+            rx_consensus,
             tx_network,
             rx_network,
-            rx_order: None,
+            rx_order: rx_order.fuse(),
             auth_cryptostore,
             store: UnitStore::new(),
             requests: BinaryHeap::new(),
-            request_ticker: time::interval(TICK_INTERVAL),
+            request_ticker: time::interval(TICK_INTERVAL).fuse(),
```

Comment:
I have just noticed that since `Environment` is no longer a future, we do not need to store `Interval` as a field -- we can create it in the `run_epoch` method.  I will fix it